Added support for ADP1051 and ADP1055 on existing ADP1050 driver

actorreno opened the pull request:
## PR Description

- This adds ADP1051 and ADP1055 driver support on existing ADP1050 Driver.
- Datasheets [ADP1051](https://www.analog.com/media/en/technical-documentation/data-sheets/ADP1051.pdf), [ADP1055](https://www.analog.com/media/en/technical-documentation/data-sheets/ADP1055.pdf).
- Tested on RPI4 using the following eval boards: [ADP1051-EVB](https://www.analog.com/media/en/technical-documentation/user-guides/UG-566.pdf), [ADP1055-EVB](https://www.analog.com/media/en/technical-documentation/user-guides/ADP1055-EVALZ_UG-710.pdf).


## PR Type
- [ ] Bug fix (a change that fixes an issue)
- [x] New feature (a change that adds new functionality)
- [ ] Breaking change (a change that affects other repos or cause CIs to fail)

## PR Checklist
- [x] I have conducted a self-review of my own code changes
- [x] I have tested the changes on the relevant hardware
- [ ] I have updated the documentation outside this repo accordingly (if there is the case)

nunojsa reviewed (commented):
Another hint is to run:

`git log --oneline drivers/hwmon/pmbus` to look for the style used in that dir for commit titles. The most common seems to be (adapted for your case):

`hwmon: (pmbus/adp1050): Support adp1051 and adp1055`

actorreno commented:
v2

.yaml:

- updated description to scale as suggested
- removed further descriptions and simply attached datasheet links
- reverted back most of the device tree sample
        (just a lack of understanding on the yaml file, i thought it was a full device tree)
        (no fallback feature intended, but will note just in case)
        
.rst
- dropped the small unrelated change

.c
- dropped the unrelated change, astyle adjusts this each run
- dopped the ID and directly passed the pmbus structs
- used the error checking as suggested

nunojsa commented on Documentation/devicetree/bindings/hwmon/pmbus/adi,adp1050.yaml:
```diff
@@ -10,16 +10,35 @@ maintainers:
+      - enum:
+          - adi,adp1050
+          - adi,adp1051
+          - adi,adp1055
```
On top of what I've said already, you don't need "items:" in here (as you have things now at least).

nunojsa commented on drivers/hwmon/pmbus/adp1050.c:
```diff
@@ -23,21 +23,57 @@ static struct pmbus_driver_info adp1050_info = {
-	return pmbus_do_probe(client, &adp1050_info);
+	struct pmbus_driver_info *info;
+
+	info = (struct pmbus_driver_info *)device_get_match_data(&client->dev);
```
no need for the cast..

actorreno replied:
I added the cast just to remove a warning but yes this can be removed to simplify the code further.

nunojsa commented on drivers/hwmon/pmbus/adp1050.c:
```diff
@@ -23,21 +23,57 @@ static struct pmbus_driver_info adp1050_info = {
 			 const struct i2c_device_id *id)
 {
-	return pmbus_do_probe(client, &adp1050_info);
+	struct pmbus_driver_info *info;
```
It can be `const pmbus_driver_info *info`

actorreno replied:
`const pmbus_driver_info *info;` returns an error similar to the one below. Will keep the original `struct pmbus_driver_info *info;`

![image](https://github.com/user-attachments/assets/5e48af5f-c24b-411b-a6ae-31d80cfb2b8e)

nunojsa replied:
Obviously a typo from me... `const struct pmbus_driver_info *info` and then no need for the cast

actorreno replied:
updated the commit but still fails some checks.

a warning remains it seems
![image](https://github.com/user-attachments/assets/5d35d136-19fb-4ca6-8623-a7b14044bf78)

nunojsa replied:
Yes, my bad... pmbus_do_probe() discards the const qualifier. Sorry for the noise.

actorreno replied:
reverted back to original

struct pmbus_driver_info *info;

nunojsa reviewed (commented):
Fell free to send this upstream...

nunojsa commented:
Please send this upstream so we can merge this here

actorreno commented:
This was already sent upstream. However, the maintainers caught this and another driver LTP8800 to be somewhat similar and suggested we combine them and send together in one patch. There was feedback on LTP8800 and am still waiting for updates on it. Will apply the recent feedback here too to the upstream when ready.

Also it seems the main branch file was updated and I merged/resolved conflicts it above. I hope this is ok. 
The main file change seems to be related to the probe() version not being 6.1 anymore

/* 6.1 probe() function still uses the second struct i2c_device_id argument */
static int adp1050_probe(struct i2c_client *client,
			 const struct i2c_device_id *id)

nunojsa commented:
> I got in touch with the one handling the other part and it seems this was the latest update:
https://lore.kernel.org/all/Z5O-avYZkWsD-Vka@smile.fi.intel.com/
I already got in touch with the other engineer to start moving this forward, the feedbacks doesn't seem too hard so I am hoping something substantial happen next week. Thanks!

Please, it looks pretty much ready to be accepted so...
